server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AbstractHistogramAggregator.java

```diff
@@ -100,7 +97,10 @@ public AbstractHistogramAggregator(
 
                 EmptyBucketInfo emptyBucketInfo = null;
                 if (minDocCount == 0) {
-                    emptyBucketInfo = new EmptyBucketInfo(interval, offset, minBound, maxBound, buildEmptySubAggregations());
+                    emptyBucketInfo = new EmptyBucketInfo(interval, offset,
+                        extendedBounds == null || extendedBounds.getMin() == null ? Double.POSITIVE_INFINITY : extendedBounds.getMin(),
```

Comment:
Nit: Could we provide a default `DoubleBounds(null, null)` in the constructor, and then just call `DoubleBounds#effectiveMin` here? (ditto for max and the below usage).  Especially since the logic for what to use for undefined is a little arbitrary (e.g. it's not obvious at a glance why we use `POSITIVE_INFINITY` instead of `NAN`), I'd rather just keep that logic in one place if possible.